### PR TITLE
[38140136] Fix email/mobile phone uniqueness when setting to empty values

### PR DESCRIPTION
**Pull request by emerencia**
Fixes https://freedcamp.com/view/2354427/tasks/panel/task/38140136

### CODE REVIEW

File: app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb [hunk 1]
```diff
@@ -28,12 +28,24 @@ def destroy
...
+
         def update_person
-          timestamp = person_params[:timestamp] and person_params[:timestamp].to_datetime
+          timestamp = person_params[:timestamp].present? ? Time.zone.parse(person_params[:timestamp]) : nil
           # no timestamp: request is synchronous
           # if timestamp: request is async, so check if it is the most recent one
-          if timestamp.nil? || (timestamp > current_user.updated_at)
-            return current_user.update(person_params.except(:timestamp))
+          if timestamp.blank? || timestamp > current_user.updated_at
+            return current_user.update(person_params.except(:timestamp).merge(overwritten_params))
```
Comment (frbl) [inline app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb:48]: do we only want to do this for an update? Or should we also do this for a create?

**Review by frbl — APPROVED** [review]
lgtm

File: app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb (hunk 1 above)
Comment (emerencia) [reply app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb:48]: That's a very good point. I'll see if it affects create as well.

Comment (emerencia) [reply app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb:48]: Ok I looked this up and the users are created as part of the auth_user.rb/JWT authentication flow, which eventually calls `CreateAnonymousUser`, which always initializes the users with a nil email and mobile_phone number.

Comment (emerencia) [reply app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb:48]: I checked all the other people_controllers and person_controllers for methods that update or create a person object. I found one more controller where it is possible to update a person's email and/or mobile_phone, I'll fix that one also.

Comment (emerencia) [reply app/controllers/api/v1/cookie_and_jwt_api/person_controller.rb:48]: Fixed.

**Pull request merged**